src/NUnitFramework/framework/Constraints/RangeConstraint.cs

```diff
@@ -72,7 +87,7 @@ public override ConstraintResult ApplyTo<TActual>(TActual actual)
         {
             if ( from == null || to == null || actual == null)
                 throw new ArgumentException( "Cannot compare using a null reference", "actual" );
-
+            CompareFromAndTo();
```

Comment:
I'd be happier if this was checked when `comparer` is changed in the `Using` methods because that catches possible bugs immediately rather than deferring until this part of the constraint is evaluated- which may be never.
That also guarantees that this check can be removed.